docs/architecture/adr/ADR043-k8s-resource-access.md

```diff
@@ -0,0 +1,530 @@
+# ADR043: Kubernetes resource access rules
+
+## Status
+
+Pending
+
+## Context
+
+Several different levels of access are required within a kubernetes cluster. The
+GSP uses role based access control so these levels are granted to users and
+groups via roles.
+
+## Decision
+
+We will create two levels of access within each namespace:
+
+* Operator
+* Auditor
+
+The Operator is a relatively permissive read-write role within the namespace.
+Developers working on branches that are not part of the release process may be
+granted this role in certain namespaces. This is also the role the in-cluster
+concourse team for each namespace will be granted.
+
+The Auditor is mostly read-only and will be given to all authenticated users in
+the cluster.
+
+The complete list of resource permissions is given in Appendix A.
+
+## Consequences
+
+* Site Reliability Engineers (SRE) will be able to effectively operate the GSP
+  while reducing the need to escalate to "admin" level access.
+* service team members (e.g. devs) will be able to iterate changes quickly via
+  branch-based development and/or `kubectl apply` level access to specific
+  namespaces.
+* service team members will not be able to interfere with other devs'
+  namespaces, "system" namespaces or "prod" namespaces.
+* SREs and service team members will not be able to cause service degradations
+  or outages to production services.
+
+
+## Appendix A: Permissions rules
+
+```
+ApiGroup: ""
```

Comment:
What format is this? It looks like YAML but it isn't because it has duplicate `ApiGroup` keys.  I'm happy to push a change to make it look more YAMLy but it still wouldn't be something we can copy-paste into a `Role` or `ClusterRole` config.

I think it should either be machine-readable, or it shouldn't look machine-readable.  So I'd prefer we either made it proper kubeyaml (or a subtree of kubeyaml), or format it using markdown instead of a big preformatted block.

Reply:
My original ambition with this was to make it as close to kubeyaml as possible, but still make the comparison between the two easy. That turned out not to be the case at all when I actually created the roles. So we can reformat this as markdown; but I've no idea how that should be formatted. Did you have a layout in mind?

Reply:
Or...I could just paste in the now-created `ClusterRole` definitions?

Reply:
I pushed a commit formatting it as markdown, but I'd be happy if you pasted in your ClusterRole defns as well. Don't mind either way.